scripts/packageComponent.js

```diff
@@ -10,62 +11,130 @@ const replace = require('replace-in-file')
 
 const {generateCRXFile, installErrorHandlers} = require('./lib/util')
 
-function stageFiles (commander, outputDir, datFile) {
-  const baseDatFile = path.parse(datFile).base
+function stageFiles (datFile, manifestsDir, outputDir) {
+  const parsedDatFile = path.parse(datFile)
 
-  const originalManifest = path.join('manifests', commander.type + '-manifest.json')
-  const updatedManifest = path.join(outputDir, 'manifest.json')
+  const datFileBase = parsedDatFile.base
+  const datFileName = getNormalizedDATFileName(parsedDatFile.name)
+
+  const originalManifest = path.join(manifestsDir, `${datFileName}-manifest.json`)
+  const outputManifest = path.join(outputDir, 'manifest.json')
 
   const replaceOptions = {
-    files: updatedManifest,
-    from: /0\.0\.0/g,
+    files: outputManifest,
+    from: /0\.0\.0/,
     to: commander.setVersion
   }
 
   mkdirp.sync(outputDir)
 
-  fs.copyFileSync(originalManifest, updatedManifest)
-  fs.copyFileSync(datFile, path.join(outputDir, baseDatFile))
+  fs.copyFileSync(originalManifest, outputManifest)
+  fs.copyFileSync(datFile, path.join(outputDir, datFileBase))
 
   replace.sync(replaceOptions)
 }
 
-function getDATFileFromComponentType (componentType) {
+function generateManifestFilesByComponentType (componentType) {
   switch (componentType) {
     case 'ad-block-updater':
-      return path.join('node_modules', 'ad-block', 'out', 'ABPFilterParserData.dat')
+      childProcess.execSync(`npm run --prefix ${path.join('node_modules', 'ad-block')} manifest-files`)
+      break
     case 'tracking-protection-updater':
-      return path.join('node_modules', 'tracking-protection', 'data', 'TrackingProtection.dat')
+      // TODO(emerick): Make this work like ad-block (i.e., update the
+      // tracking-protection repo with a script to generate the
+      // manifest and then call that script here)
+      break
     default:
       throw new Error('Unrecognized component extension type: ' + componentType)
   }
 }
 
+function getManifestsDirByComponentType (componentType) {
+  switch (componentType) {
+    case 'ad-block-updater':
+      return path.join('node_modules', 'ad-block', 'out')
+    case 'tracking-protection-updater':
+      // TODO(emerick): Make this work like ad-block
+      return path.join('manifests')
+    default:
+      throw new Error('Unrecognized component extension type: ' + componentType)
+  }
+}
+
+function getNormalizedDATFileName (datFileName) {
+  if (datFileName === 'ABPFilterParserData' || datFileName === 'TrackingProtection') {
+    return 'default'
+  }
+  return datFileName
+}
+
+function getDATFileListByComponentType (componentType) {
+  let list = []
+
+  switch (componentType) {
+    case 'ad-block-updater':
+      fs.readdirSync(path.join('node_modules', 'ad-block', 'out'))
+        .filter(file => {
+          return (path.extname(file) === '.dat' && file !== 'SafeBrowsingData.dat')
+        })
+        .forEach(file => {
```

Comment:
I'm ok with this but just noting you could do a reduce after the filter to achieve the same slightly more cleanly.